accessible/toolbox_treeview.component.js

```diff
@@ -0,0 +1,189 @@
+      <ol role='group' *ngIf='displayBlockMenu'  [attr.aria-level]='level+2'>
+        <li #workspaceCopy id='{{treeService.createId(workspaceCopy)}}' role='treeitem' aria-selected=false [attr.aria-level]='level+2'>
+          <button #workspaceCopyButton id='{{treeService.createId(workspaceCopyButton)}}' (click)='copyToWorkspace(block)'>copy to workspace</button>
+          {{setLabelledBy(workspaceCopy, concatStringWithSpaces(workspaceCopyButton.id, 'blockly-button'))}}
+        </li>
+        <li #blockCopy id='{{treeService.createId(blockCopy)}}' role='treeitem' aria-selected=false [attr.aria-level]='level+2'>
+          <button #blockCopyButton id='{{treeService.createId(blockCopyButton)}}' (click)='sharedClipboardService.copy(block)'>copy to clipboard</button>
+          {{setLabelledBy(blockCopy, concatStringWithSpaces(blockCopyButton.id, 'blockly-button'))}}
+        </li>
+        <li #sendToSelected id='{{treeService.createId(sendToSelected)}}' role='treeitem' aria-selected=false [attr.aria-level]='level+2'>
+          <button #sendToSelectedButton id='{{treeService.createId(sendToSelectedButton)}}' (click)='copyToMarked(block)' disabled='{{notCompatibleWithMarkedBlock(block)}}' [attr.aria-disabled]='notCompatibleWithMarkedBlock(block)'>copy to marked spot</button>
+          {{setLabelledBy(sendToSelected, concatStringWithSpaces(sendToSelectedButton.id, 'blockly-button', notCompatibleWithMarkedBlock(block)))}}
+        </li>
+      </ol>
+      {{addClass(listItem, 'blocklyHasChildren')}}
+    </li>
+    <div *ngFor='#inputBlock of block.inputList; #i=index'>
+      <field-view [attr.aria-level]='level+1' *ngFor='#field of getInfo(inputBlock); #j=index' [field]='field' [level]='level+1'></field-view>
+      <toolbox-tree-view *ngIf='inputBlock.connection && inputBlock.connection.targetBlock()' [block]='inputBlock.connection.targetBlock()' [displayBlockMenu]='false' [level]='level+1'></toolbox-tree-view>
+      <li aria-selected=false #listItem1 role='treeitem' [attr.aria-level]='level+1' id='{{treeService.createId(listItem1)}}' *ngIf='inputBlock.connection && !inputBlock.connection.targetBlock()'>
+        <label #label id='{{treeService.createId(label)}}'>{{inputType(inputBlock.connection)}} {{valueOrStatement(inputBlock)}} needed:</label>
+        {{setLabelledBy(listItem1, concatStringWithSpaces('blockly-argument-text', label.id))}}
+      </li>
+    </div>
+  </ol>
+</li>
+<toolbox-tree-view *ngIf= 'block.nextConnection && block.nextConnection.targetBlock()' [level]='level' [block]='block.nextConnection.targetBlock()' [displayBlockMenu]='false'></toolbox-tree-view>
+    `,
+    directives: [ng.core.forwardRef(
+        function() { return blocklyApp.ToolboxTreeView; }), blocklyApp.FieldView],
+    inputs: ['block', 'displayBlockMenu', 'level', 'index', 'tree'],
+  })
+  .Class({
+    constructor: [blocklyApp.ClipboardService, blocklyApp.TreeService, function(_service, _service2) {
```

Comment:
Because most official Angular2 documentation is in typescript, I only learned about this syntax from  people's blogs. So I take it back, it's probably not standard Angular practice.